warehouse/integrations/middleware/sql.go

```diff
@@ -0,0 +1,124 @@
+package middleware
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/warehouse/logfield"
+)
+
+const ExecutingQuery = "executing query"
+
+type Opt func(*DB)
+
+type DB struct {
+	*sql.DB
+
+	since              func(time.Time) time.Duration
+	logger             logger.Logger
+	keysAndValues      []any
+	slowQueryThreshold time.Duration
+	secretsRegex       map[string]string
+}
+
+func WithLogger(logger logger.Logger) Opt {
+	return func(s *DB) {
+		s.logger = logger
+	}
+}
+
+func WithKeyAndValues(keyAndValues ...any) Opt {
+	return func(s *DB) {
+		s.keysAndValues = keyAndValues
+	}
+}
+
+func WithSince(since func(time.Time) time.Duration) Opt {
```

Comment:
We are using this while calculating the execution time. Since execution time cannot be predicted during test, providing a mocked version during tests.

Reply:
We are using this while calculating the execution time. Since execution time cannot be predicted during test, providing a mocked version during tests.

Reply:
Keep in mind that in your tests you can access unexported symbols so you could just do:

```go
db := &DB{
    since: func() { /* something */ }
}
// use db
```

This way we could remove the `WithSince` option (and possibly others) that don't add much value to the end user of this package.

Reply:
This would require me to have the test with the same package right instead of `sqlquerywrapper_test`?